### PR TITLE
New package: mysql8-8.0.21

### PR DESCRIPTION
**Pull request by rien**
This PR adds the MySQL database version 8.

#### Motivation

The other `mysql` package is still version 5.6, but the reason I created a new package is because the upgrade would break most databases since MySQL only has backwards compatibility for one major version. Instead of creating a lot of frustration I took the path void has already taken with postgresql (e.g. creating the `postgresql12` package next to `postgresql`).

#### LibreSSL support

MySQL does not support libressl by default, but I've used some patches from [gentoo](https://dev.gentoo.org/~whissi/dist/mysql/) (with permission of @Whissi). They also maintain a repository with working changes over at https://github.com/Whissi/gentoo-mysql.

#### Cross compiling

I gave up cross compiling this package because I've already spent quite some time getting this compile on x86. Sorry.

During configuration it tries to detect library versions by compiling and running small binaries, but this won't work when cross compiling. I've tried setting `build_helper=qemu` but it dies without error message, just error code 2. At that point i gave up because building this package takes ages.

#### Initialization

I've tried to replicate the behavior of the original `mysql` package as close as possible. The `INSTALL` script initializes the data directory and sets an empty root password. It also echo's a message telling the user they should run `mysql_secure_installation` in order to set a root password and to remove test databases and anonymous users.

The given configuration only allows connections from localhost, so the security risk is at a minimum here. However, if desired I can let the initialization step generate a random root password.

### CODE REVIEW

**Comment by rien** [comment]
@Vaelatern in 48c311e12c28d7fc13e5efc7073fc312c4815e4b (4 years ago...) you noticed the change in signature of `CRYPTO_set_mem_functions`. Do you have any idea how to solve this in this case? Is there some dummy value I can pass?

**Comment by rien** [comment]
Nevermind, I've found the Gentoo patchset which looks promising.

File: srcpkgs/mysql8/INSTALL [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+case ${ACTION} in
+post)
+	install -dm0700 var/lib/mysql
+	bin/mysqld --user=mysql --initialize-insecure --datadir=/var/lib/mysql
+	chown -R mysql:mysql var/lib/mysql
+	echo "Your MySQL password is empty."
+	echo "Please secure your database by running"
+	echo "# mysql_secure_database"
+	;;
+esac
```
Comment (Vaelatern) [inline srcpkgs/mysql8/INSTALL:12]: Why do this initialization for the user?

File: srcpkgs/mysql8/files/my.cnf [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+# but will not function as a master if omitted
+server-id	= 1
+
+[mysqldump]
+quick
+max_allowed_packet = 16M
+
+[mysql]
+# Remove the next comment character if you are not familiar with SQL
+#safe-updates
```
Comment (Vaelatern) [inline srcpkgs/mysql8/files/my.cnf:42]: Why a custom config file?

File: srcpkgs/mysql8/patches/0002-cmake-Remove-Werror.patch [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+-  STRING_APPEND(MY_C_WARNING_FLAGS   " -Werror")
+-  STRING_APPEND(MY_CXX_WARNING_FLAGS " -Werror")
+-ENDIF()
+-
+ # Set warning flags for gcc/g++/clang/clang++
+ IF(MY_COMPILER_IS_GNU_OR_CLANG)
+   STRING_APPEND(CMAKE_C_FLAGS   " ${MY_C_WARNING_FLAGS}")
+-- 
+2.27.0
+
```
Comment (Vaelatern) [inline srcpkgs/mysql8/patches/0002-cmake-Remove-Werror.patch:30]: Why not turn off maintainer mode.

**Comment by ericonr** [comment]
PR has been inactive and since then we've switched over to OpenSSL and MariaDB. Closing for now.

**Pull request closed without merging**